core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -5476,7 +5475,8 @@ private void validateSnapshot(
       SqlSnapshot snapshot = (SqlSnapshot) node;
       SqlNode period = snapshot.getPeriod();
       RelDataType dataType = deriveType(requireNonNull(scope, "scope"), period);
-      if (dataType.getSqlTypeName() != SqlTypeName.TIMESTAMP) {
+      if (dataType.getSqlTypeName() != SqlTypeName.TIMESTAMP
```

Comment:
This is done as a workaround to fix the tests.

Reply:
I'm fine with this workaround, but can we add a comment here to this effect? I'm a bit afraid that we'll forget that we've ever made this change if we don't mark it in some way.